### PR TITLE
Add DXF import to the OCP branch

### LINKED ISSUE #148: Fluent API does not allow to easily extrude an existing face
It is possible with the following (not so fluent) workaround:

```python
cap_m = cap.cut(cut_block,combine=False).faces('<Z').wires()
cap_m.ctx.pendingWires = cap_m.vals()
cap_m = cap_m.extrude(-h3,combine=True)
```

This could be solved by e.g. adding an additional function like `toPendingWires()` which would append current `vals()` to the `pendingWires` field.

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Preliminary support for DXF importing. Entities are converted to faces. So far only lines, polylines, arcs and circles are supported.

NB: to enable further processing `toPending` is also implemented in this PR, so it will resolve #148 

Example usage:
```python
res = importers.importDXF('gear.dxf',1e-3).wires().toPending().extrude(5)
```

obraz

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
@RustyVermeer you did some work on DXF import, any thoughts on this PR?

File: cadquery/cq.py [hunk 1]
```diff
@@ -3441,6 +3441,18 @@ def section(self, height=0.0):
 
         return self.newObject([r])
 
+    def toPending(self):
+        """
+        Adds wires/edges to pendingWires/pendingEdges.
+        
+        :return: same CQ object wit updated context.
```
Comment (jmwright) [inline cadquery/cq.py:3448]: wit should be with

File: cadquery/occ_impl/importers.py [hunk 2]
```diff
@@ -53,3 +63,93 @@ def importStep(fileName):
...
+            edges_in.Append(e.wrapped)
+        ShapeAnalysis_FreeBounds.ConnectEdgesToWires_s(edges_in, tol, False, wires_out)
+
+        rv = [Shape.cast(el) for el in wires_out]
+
+    return rv
+
+
+def importDXF(filename, tol=1e-6):
+
```
Comment (jmwright) [inline cadquery/occ_impl/importers.py:144]: No docstring?

File: tests/test_importers.py [hunk 3]
```diff
@@ -6,13 +6,15 @@
 import os
 
 from cadquery import *
-from cadquery import exporters
 from cadquery import importers
```
Comment (jmwright) [inline tests/test_importers.py:9]: Doesn't the `from cadquery import *` line above make this redundant?

Comment (adam-urbanczyk) [reply tests/test_importers.py]: Corrected

**Comment by adam-urbanczyk** [comment]
Alright, I think all the issues are solved.

**Comment by jmwright** [comment]
+1 for merge

**Comment by adam-urbanczyk** [comment]
OK, merging.

**Comment by adam-urbanczyk** [comment]
LOL, merged to he wrong branch (OCP instead of master).

**Comment by lalebarde** [comment]
Hi Adam, I am trying make use of `toPending()` to extrude a face, but my use of it looks bad:

```
decoupe = cq.Workplane("XY").rect(10, 20, centered=True).extrude(100)
decoupe = decoupe('<Y').wires().toPending().extrude(-100)
```

It results in the following error:

`TypeError: 'Worplane' object is not callable`

**Comment by adam-urbanczyk** [comment]
You probably meant:
```python
decoupe = cq.Workplane("XY").rect(10, 20, centered=True).extrude(100)
decoupe = decoupe.faces('<Y').wires().toPending().extrude(-100)
```

**Comment by lalebarde** [comment]
Oups, thank you Adam

**Pull request merged**